editor/src/core/webpack-loaders/css-loader.ts

```diff
@@ -1,15 +1,68 @@
 import { LoadModule, loadModuleResult, MatchFile, ModuleLoader } from './loader-types'
 
+export const InjectedCSSFilePrefix = 'injected-css-file-'
+
 const matchFile: MatchFile = (filename: string) => {
   return ['.css'].some((extension) => filename.endsWith(extension))
 }
 
 const loadModule: LoadModule = (filename: string, contents: string) => {
+  const loadedContents = `
+    Object.defineProperty(module, 'exports', {
+      get() {
+        // this side effect will run every time the module is required
+        (function() {
+          'use strict';
+          const filename = ${JSON.stringify(filename)}
+          const content = ${JSON.stringify(contents)}
+          const elementId = ${JSON.stringify(InjectedCSSFilePrefix)} + filename;
+          const maybeExistingTag = document.getElementById(elementId);
+          if (maybeExistingTag != null) {
+            if (maybeExistingTag.textContent === content) {
+              return;
+            } else {
+              maybeExistingTag.parentElement.removeChild(maybeExistingTag);
+            }
+          }
+          const styleTag = document.createElement("style");
+          styleTag.type = "text/css";
+          styleTag.id = elementId;
+          styleTag.appendChild(document.createTextNode(content));
+          document.querySelector("head").appendChild(styleTag);
+        })();
+        return {};
+      },
+    })
+  `
+
   // FIXME Replace CSS custom evaluation from evaluator.ts with this
-  return loadModuleResult(filename, contents)
+  return loadModuleResult(filename + '.js', loadedContents)
 }
 
 export const CSSLoader: ModuleLoader = {
   match: matchFile,
   load: loadModule,
 }
+
+// FIXME This doesn't quite fit in here. We need a generalised strategy for unloading files when they're removed from the list of imports
+export function unimportAllButTheseCSSFiles(filesToKeep: Array<string>): void {
+  const head = document.querySelector('head')
+  const shouldBeRemoved = (elemId: string) =>
```

Comment:
There's a danger (albeit a small one) of this deleting something with a filename that ends the same as another.
`/style.css` versus `/home/style.css` for example.

Reply:
I've just double checked and this isn't an issue because it uses the full path to the file, not just the file name.